src/Portal.sol

```diff
@@ -58,9 +71,72 @@ abstract contract Portal is NttManagerNoRateLimiting, IPortal {
         return _currentIndex();
     }
 
-    /* ============ Internal Interactive Functions ============ */
+    /* ============ External Interactive Functions ============ */
+
+    /// @inheritdoc IPortal
+    function setDestinationWrappedMToken(
+        address sourceWrappedToken_,
+        uint16 destinationChainId_,
+        bytes32 destinationWrappedToken_
+    ) external onlyOwner {
+        if (destinationChainId_ == chainId) revert InvalidDestinationChain(destinationChainId_);
+
+        destinationWrappedMToken[sourceWrappedToken_][destinationChainId_] = destinationWrappedToken_;
+        emit DestinationWrappedMTokenSet(sourceWrappedToken_, destinationChainId_, destinationWrappedToken_);
+    }
+
+    /// @inheritdoc IPortal
+    function transferWrappedMToken(
+        uint256 amount_,
+        address sourceWrappedToken_,
+        uint16 destinationChainId_,
+        bytes32 recipient_,
+        bytes32 refundAddress_
+    ) external payable returns (bytes32 messageId_) {
+        if (amount_ == 0) revert ZeroAmount();
+        if (recipient_ == bytes32(0)) revert InvalidRecipient();
+        if (refundAddress_ == bytes32(0)) revert InvalidRefundAddress();
+
+        bytes32 destinationWrappedToken_ = destinationWrappedMToken[sourceWrappedToken_][destinationChainId_];
+
+        if (destinationWrappedToken_ == bytes32(0))
+            revert UnsupportedDestinationToken(sourceWrappedToken_, destinationChainId_);
+
+        // transfer Wrapped M from the sender
+        IERC20(sourceWrappedToken_).transferFrom(msg.sender, address(this), amount_);
+
+        // unwrap Wrapped M token to M Token
+        amount_ = IWrappedMTokenLike(sourceWrappedToken_).unwrap(address(this), amount_);
```

Comment:
Since we will be whitelisting the supported wrapped tokens, it seems safe to rely on the `amount` returned by `unwrap`.
Ideally we should retrieve the balance in the contract before, then after and compute the exact amount that was unwrapped.